src/CamadaFisica.cpp

```diff
@@ -7,18 +7,38 @@
 using namespace std;
 
 vector<int> CamadaFisicaTransmissoraCodificacaoBinaria(vector<int> quadro) {
+    //a função CamadaDeAplicacaoTransmissora já transforma em Binário
     return quadro;
 }
 
 vector<int> CamadaFisicaTransmissoraCodificacaoBipolar(vector<int> quadro) {
     vector<int> result;
     vector<int>::iterator iter = quadro.begin();
-
+/*
     while (iter < quadro.end()) {
         result.insert(result.begin(), *iter > 0? 1 : -1);
         iter++;
     }
 
+    return result;
+    */
+    // vê se o ultimo 1 armazenado foi positivo ou negativo, 0 == negativo, 1 == positivo
+    int anterior = 0;
+    while (iter < quadro.end()) {
```

Comment:
Será que conseguimos utilizar um for loop ao invés de um while aqui? Acredito que facilitaria a leitura.

Reply:
pode ser.